Feature database creation

allenhinson877 opened the pull request:
Implemented functions to allow upload of data to our Firestore database. We are able to upload message, create user, create group, and fetch messages by group and display them.

allenhinson877 committed:
Database creation

allenhinson877 committed:
databasefacade.proto implementation

allenhinson877 committed:
database.py update

Issue with some of the declarations from last one

allenhinson877 committed:
database.py update

allenhinson877 committed:
database.py update

Updated functions to reflect to updated globalmessage.proto

allenhinson877 committed:
update

Added functionality to check if a user or group is already in the database. Also added functionality for users to be stored by username in a collection for the group they are chatting in.

allenhinson877 committed:
update

Update consistency between proto file and .py file

allenhinson877 committed:
Update database.py

Fixed issue with some of the functions after testing them with random data to make sure they worked correctly.

scottcalebc commented:
Should close #9 and #8

scottcalebc reviewed (changes requested):
I would remove interface/__init__.py and the generated pb2 files from tracking, the init.py was a holdover from a previous attempt at solving the import issue in the src files. The pb2 python files are generated by the python protoc compiler and do not need to be tracked

allenhinson877 committed:
interfaces clean up

Removed init.py and ran ./buildProto clean interfaces to get rid of proto files

scottcalebc reviewed (approved):
Looks good!
